### ISSUE TITLE
Test failure

### ISSUE THREAD

**Issue by mohawk2** [chorny/Win32-TieRegistry#1]
perl -V:

```
Summary of my perl5 (revision 5 version 20 subversion 0) configuration:

  Platform:
    osname=MSWin32, osvers=6.2, archname=MSWin32-x64-multi-thread
    uname='Win32 strawberry-perl 5.20.0.1 #1 Fri May 30 08:27:00 2014 x64'
    config_args='undef'
    hint=recommended, useposix=true, d_sigaction=undef
    useithreads=define, usemultiplicity=define
    use64bitint=define, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags =' -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE  -DPERL_TEXTMODE_
SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-st
rict-aliasing -mms-bitfields',
    optimize='-s -O2',
    cppflags='-DWIN32'
    ccversion='', gccversion='4.8.2', gccosandvers=''
    intsize=4, longsize=4, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long long', ivsize=8, nvtype='double', nvsize=8, Off_t='long long',
lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='g++', ldflags ='-s -L"D:\straw(berry) perl\perl\lib\CORE" -L"D:\straw(be
rry) perl\c\lib"'
    libpth=D:\straw(berry) perl\c\lib D:\straw(berry) perl\c\x86_64-w64-mingw32\
lib D:\straw(berry) perl\c\lib\gcc\x86_64-w64-mingw32\4.8.2
    libs=-lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32
 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm -lversion
 -lodbc32 -lodbccp32 -lcomctl32
    perllibs=-lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladva
pi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm -lver
sion -lodbc32 -lodbccp32 -lcomctl32
    libc=, so=dll, useshrplib=true, libperl=libperl520.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_win32.xs, dlext=xs.dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags='-mdll -s -L"D:\straw(berry) perl\perl\lib\CORE" -
L"D:\straw(berry) perl\c\lib"'


Characteristics of this binary (from libperl):
  Compile-time options: HAS_TIMES HAVE_INTERP_INTERN MULTIPLICITY
                        PERLIO_LAYERS PERL_DONT_CREATE_GVSV
                        PERL_HASH_FUNC_ONE_AT_A_TIME_HARD
                        PERL_IMPLICIT_CONTEXT PERL_IMPLICIT_SYS
                        PERL_MALLOC_WRAP PERL_NEW_COPY_ON_WRITE
                        PERL_PRESERVE_IVUV USE_64_BIT_INT USE_ITHREADS
                        USE_LARGE_FILES USE_LOCALE USE_LOCALE_COLLATE
                        USE_LOCALE_CTYPE USE_LOCALE_NUMERIC USE_PERLIO
                        USE_PERL_ATOF
  Built under MSWin32
  Compiled at May 30 2014 08:30:58
  @INC:
    D:/straw(berry) perl/perl/site/lib/MSWin32-x64-multi-thread
    D:/straw(berry) perl/perl/site/lib
    D:/straw(berry) perl/perl/vendor/lib
    D:/straw(berry) perl/perl/lib
    .
```

Log:

```
    "D:\straw(berry) perl\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t\*.t
t\01_compile.t .. ok

#   Failed test 'Opened CU/SW/MS/Win/CV/Pol/Exp//NoDriveTypeAutoRun'
#   at t\02_main.t line 26.
# $^E = The handle is invalid

#   Failed test 'Type is REG_DWORD'
#   at t\02_main.t line 27.
#          got: '4'
#     expected: undef

#   Failed test 'Value matches expected'
#   at t\02_main.t line 28.
#                   undef
#     doesn't match '(?^i:^0x[\da-f]{8}$)'
# Looks like you failed 3 tests of 3.
t\02_main.t ..... 
Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/3 subtests 

Test Summary Report
-------------------
t\02_main.t   (Wstat: 768 Tests: 3 Failed: 3)
  Failed tests:  1-3
  Non-zero exit status: 3
Files=2, Tests=5,  0 wallclock secs ( 0.03 usr +  0.05 sys =  0.08 CPU)
Result: FAIL
Failed 1/2 test programs. 3/5 subtests failed.
NMAKE : fatal error U1077: '"D:\straw(berry) perl\perl\bin\perl.exe"' : return code '0x3'
```

**Comment by Bitdreamr** [comment]
This was reported on CPAN https://rt.cpan.org/Public/Bug/Display.html?id=102385 and a code fix was described.
ActiveState is refusing to take version 0.28 because they are labeling this test bug as a blocker.
Please apply this fix as soon as possible!

After `":REG_"`, add `":KEY_"`,
Then before `my $val` add this:
`$reg = $reg->Open('', {Access => KEY_READ} );`

**Comment by Bitdreamr** [comment]
Note that this bug does not occur in a window with Administrator privileges.  In a non-admin window the registry must be opened ReadOnly.

**Comment by mohawk2** [comment]
@Bitdreamr, could you make a PR with this change?

**Comment by chorny** [comment]
Was fixed in 0.29
